File: clients/csharp/examples/tables-list-types.cs
using System.Text.Json;
using DataStax.AstraDB.DataApi;

namespace Examples;

public class Program
{
  static async Task Main()
  {
    // Get an existing table
    var client = new DataAPIClient();
    var database = client.GetDatabase(
      "**API_ENDPOINT**",
      "**APPLICATION_TOKEN**"
    );

    // List types
    var types = await database.ListTypesAsync();

    Console.WriteLine(JsonSerializer.Serialize(types));
  }
}

File: clients/go/examples/tables-list-types.go
package main

import (
	"context"
	"encoding/json"
	"fmt"
	"log"

	"github.com/datastax/astra-db-go/v2/astra"
	"github.com/datastax/astra-db-go/v2/astra/options"
)

func main() {
	ctx := context.Background()

	// Get a database
	client := astra.NewClient()

	database := client.Database(
		"**API_ENDPOINT**",
		options.API().SetToken("**APPLICATION_TOKEN**"),
	)

	// List type metadata
	types, err := database.ListTypes(ctx)

	if err != nil {
		log.Fatal(err)
	}

	output, err := json.Marshal(types)
	if err != nil {
		log.Fatal(err)
	}
	fmt.Println(string(output))
}
